REGRESSION(262797@main): Menu permanently opened on github.com

carlosgcampos opened the pull request:
#### 476e8fbe38b048c71cec9587579ad59d1d8dbb32
<pre>
REGRESSION(262797@main): Menu permanently opened on github.com
<a href="https://bugs.webkit.org/show_bug.cgi?id=258290">https://bugs.webkit.org/show_bug.cgi?id=258290</a>

Reviewed by Michael Catanzaro.

In 262797@main popover attribute was disabled for apple apps linkking to
the older SDKs, but also for non-apple ports. What happens in github now
is that the popover is never hidden because display: revert ends up in
block. This is because the popover css, which sets display: none when
popover-open is not present, is not included for any element.

* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultPopoverAttributeEnabled): Enable it by default for non-apple ports again.

Canonical link: <a href="https://commits.webkit.org/266510@main">https://commits.webkit.org/266510@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6a83f3eaa93c27dca0ed58923333d11d64bd255b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14011 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/14326 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14661 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15749 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/13298 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16835 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/14410 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15968 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14181 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14772 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11871 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16457 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/12058 "Passed tests") | | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/19664 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/13134 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/12796 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/16006 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/13346 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/11209 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12621 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3395 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16955 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13190 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (approved):
My hero!

fyi: @nt1m

carlosgcampos committed:
REGRESSION(262797@main): Menu permanently opened on github.com
https://bugs.webkit.org/show_bug.cgi?id=258290

Reviewed by Michael Catanzaro.

In 262797@main popover attribute was disabled for apple apps linkking to
the older SDKs, but also for non-apple ports. What happens in github now
is that the popover is never hidden because display: revert ends up in
block. This is because the popover css, which sets display: none when
popover-open is not present, is not included for any element.

* Source/WebKit/Shared/WebPreferencesDefaultValues.cpp:
(WebKit::defaultPopoverAttributeEnabled): Enable it by default for non-apple ports again.

Canonical link: https://commits.webkit.org/266510@main
